provider/cluster_waiter_resource.go

```diff
@@ -123,12 +97,43 @@ func (r *ClusterWaiterResource) Create(ctx context.Context,
 
 func (r *ClusterWaiterResource) Read(ctx context.Context, request tfsdk.ReadResourceRequest,
 	response *tfsdk.ReadResourceResponse) {
-	// Do Nothing
+	state := &ClusterWaiterState{}
+	diags := request.State.Get(ctx, state)
+	response.Diagnostics.Append(diags...)
+	if response.Diagnostics.HasError() {
+		return
+	}
+	state, err := r.startPolling(ctx, state)
+
+	if err != nil {
+		response.Diagnostics.AddError("Can't poll cluster state", err.Error())
```

Comment:
Please add here information if this is during create or update
Something like:
```suggestion
		response.Diagnostics.AddError("Can't poll cluster state (Read resource)", err.Error())
```

Reply:
Not relevant. No read operations.